packages/editor/src/components/provider/index.js

```diff
@@ -301,15 +323,15 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 
 		// Sets the right rendering mode when loading the editor.
 		useEffect( () => {
```

Comment:
Maybe you can try using useLayoutEffect here to see if it improves the "unstyled flashing"

Reply:
Even with the layout effect, I still get this 🤔 

https://github.com/user-attachments/assets/7bb7f81a-2d22-413a-b5d6-d2fec27b997c

Reply:
I pushed a fixed for this. The issue was elsewhere, we were using the "post" blocks if the "template" didn't load yet.

Reply:
The fix works beautifully!